modules/imgcodecs/test/test_exif.cpp

```diff
@@ -9,7 +9,144 @@
 
 namespace opencv_test { namespace {
 
-/**
+    static Mat makeCirclesImage(Size size, int type, int nbits)
+    {
+        Mat img(size, type);
+        img.setTo(Scalar::all(0));
+        RNG& rng = theRNG();
+        int maxval = (int)(1 << nbits);
+        for (int i = 0; i < 100; i++) {
+            int x = rng.uniform(0, img.cols);
+            int y = rng.uniform(0, img.rows);
+            int radius = rng.uniform(5, std::min(img.cols, img.rows) / 5);
+            int b = rng.uniform(0, maxval);
+            int g = rng.uniform(0, maxval);
+            int r = rng.uniform(0, maxval);
+            circle(img, Point(x, y), radius, Scalar(b, g, r), -1, LINE_AA);
+        }
+        return img;
+    }
+
+    static std::vector<uchar> getSampleExifData() {
+        return {
+            'M', 'M', 0, '*', 0, 0, 0, 8, 0, 10, 1, 0, 0, 4, 0, 0, 0, 1, 0, 0, 5,
+            0, 1, 1, 0, 4, 0, 0, 0, 1, 0, 0, 2, 208, 1, 2, 0, 3, 0, 0, 0, 1,
+            0, 10, 0, 0, 1, 18, 0, 3, 0, 0, 0, 1, 0, 1, 0, 0, 1, 14, 0, 2, 0, 0,
+            0, '"', 0, 0, 0, 176, 1, '1', 0, 2, 0, 0, 0, 7, 0, 0, 0, 210, 1, 26,
+            0, 5, 0, 0, 0, 1, 0, 0, 0, 218, 1, 27, 0, 5, 0, 0, 0, 1, 0, 0, 0,
+            226, 1, '(', 0, 3, 0, 0, 0, 1, 0, 2, 0, 0, 135, 'i', 0, 4, 0, 0, 0,
+            1, 0, 0, 0, 134, 0, 0, 0, 0, 0, 3, 144, 0, 0, 7, 0, 0, 0, 4, '0', '2',
+            '2', '1', 160, 2, 0, 4, 0, 0, 0, 1, 0, 0, 5, 0, 160, 3, 0, 4, 0, 0,
+            0, 1, 0, 0, 2, 208, 0, 0, 0, 0, 'S', 'a', 'm', 'p', 'l', 'e', ' ', '1', '0',
+            '-', 'b', 'i', 't', ' ', 'i', 'm', 'a', 'g', 'e', ' ', 'w', 'i', 't', 'h', ' ',
+            'm', 'e', 't', 'a', 'd', 'a', 't', 'a', 0, 'O', 'p', 'e', 'n', 'C', 'V', 0, 0,
+            0, 0, 0, 'H', 0, 0, 0, 1, 0, 0, 0, 'H', 0, 0, 0, 1
+        };
+    }
+
+    static std::vector<uchar> getSampleXmpData() {
+        return {
+            '<','x',':','x','m','p','m','e','t','a',' ','x','m','l','n','s',':','x','=',
+            '"','a','d','o','b','e',':','x','m','p','"','>',
+            '<','x','m','p',':','C','r','e','a','t','o','r','T','o','o','l','>',
+            'O','p','e','n','C','V',
+            '<','/','x','m','p',':','C','r','e','a','t','o','r','T','o','o','l','>',
+            '<','/','x',':','x','m','p','m','e','t','a','>',0
+        };
+    }
+
+    static std::vector<uchar> getSampleIccpData() {
```

Comment:
Why do you use different ICCP  metadata generators for WEBP and PNG? I would expect it's the same for different formats, is not it?

Reply:
No different data was intended; it was just temporary test data and has been fixed in the latest commit.